### PR TITLE
Add multi-arch support (arm64)

### PR DESCRIPTION
**Pull request by cytopia**
# Add multi-arch support (arm64)

This PR adds Ansible images with `arm64` support.

It superseeds the following open PR's
* Closes: https://github.com/cytopia/docker-ansible/pull/99
* Closes: https://github.com/cytopia/docker-ansible/pull/98 (CC @morganchristiansson)
* Closes: https://github.com/cytopia/docker-ansible/pull/74 (CC @morganchristiansson)
* Fixes: https://github.com/cytopia/docker-ansible/issues/73 (CC @morganchristiansson)

## Additional changes

* Switch to Python 3.10
* New workflows
* Dependabot
* Release Drafter
* Point in time releases (git tags) will never be rebuilt and can be used safely

## Todo

* [x] PoC of working Workflow run
* [x] Add workflow for PR's
* [x] Add workflow for nighty builds
* [x] Re-add Ansible, Helm, Kops versions
* [x] Add `arm64` versions
* [x] Update README
* [x] Cleanup Makefile

## Thanks :hand: 

Big shoutout to @morganchristiansson for the initial work on `arm64`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,210 +2,220 @@ ifneq (,)
...
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=awshelm HELM=2.11"
-	@echo "    make build ANSIBLE=2.3 FLAVOUR=awskops KOPS=1.15"
+	@echo "    make build VERSION=2.3"
+	@echo "    make build VERSION=2.3 STAGEg=tools"
+	@echo "    make build VERSION=2.3 STAGEg=infra"
+	@echo "    make build VERSION=2.3 STAGEg=azure"
+	@echo "    make build VERSION=2.3 STAGEg=aws"
+	@echo "    make build VERSION=2.3 STAGEg=awsk8s"
+	@echo "    make build VERSION=2.3 STAGEg=awshelm HELM=2.11"
+	@echo "    make build VERSION=2.3 STAGEg=awskops KOPS=1.15"
```
Comment (morganchristiansson) [inline Makefile:69]: STAGEg is typo?

Comment (cytopia) [reply Makefile:69]: No, that's the new standard from my reusable workflow. This is a "staged" build (although they are still different flavours) - meaning, they build on top of each other. I had to adjust it here or in all of my other projects.

File: Makefile [hunk 2]
```diff
@@ -2,210 +2,220 @@ ifneq (,)
 .error This Makefile requires GNU Make.
 endif
 
-# -------------------------------------------------------------------------------------------------
-# Default configuration
-# -------------------------------------------------------------------------------------------------
-.PHONY: lint build rebuild test tag pull-base-image login push enter
+# Ensure additional Makefiles are present
```
Comment (morganchristiansson) [inline Makefile:5]: Maybe include link to repo: https://github.com/devilbox/makefiles/
I was looking for the docker build part and didn't immediately see it.

File: .github/workflows/docker-multistage-build-kops.yml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      - name: "[SETUP] Install QEMU environment"
+        uses: docker/setup-qemu-action@v2
+        id: qemu
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - name: "[SETUP] Install Docker Buildx"
+        id: buildx
+        uses: docker/setup-buildx-action@v2
```
Comment (morganchristiansson) [inline .github/workflows/docker-multistage-build-kops.yml:82]: Build uses `docker build` and not buildx so likely this one isn't needed.

qemu+binfmt misc above should be enough.

**Comment by morganchristiansson** [comment]
I see github actions runtime is 2+ hours... This is why I added docker build cache in my PR. Tho a bit hackish it worked very well to bring time for repeat runs down.

Native arm64 workers would do wonders, github actions run on azure cloud. Azure recently added arm64 instances so fingers crossed it will be available as github hosted runners in the near future. Self-hosting arm64 runners is also possible but comes with extra work/costs to host them.

File: Makefile (hunk 1 above)
Comment (cytopia) [reply Makefile:69]: :facepalm: Didn't even see the `g` in your comment. Got it.

**Comment by morganchristiansson** [comment]
I see you're using github actions artifacts to store docker images! Nice I didn't even think of doing that. Could maybe be used for build cache too. But if I understand correct you want full uncached builds sometimes too.

**Comment by cytopia** [comment]
@morganchristiansson thanks for the review! I will address your points in a follow-up PR.

**Pull request merged**